src/shared/efivars.c

```diff
@@ -412,9 +412,13 @@ int efi_get_boot_option(
 
         if (header->path_len > 0) {
                 uint8_t *dbuf;
-                size_t dnext;
+                size_t dnext, doff;
+
+                doff = offsetof(struct boot_option, title) + title_size;
+                dbuf = buf + doff;
```

Comment:
maybe move this line after the if check below. I mean, the if check validates `doff`, hence it makes semantical sense if we'd not use `doff` before the validity check was successful. yes, given that we never dereference dbuf this doesn't really mattrer, but it is easy to the eye, to see the check followed by the use, and not vice versa